enterprise/com/src/main/java/org/neo4j/com/storecopy/ToFileStoreWriter.java

```diff
@@ -23,46 +23,77 @@
 import java.io.IOException;
 import java.io.RandomAccessFile;
 import java.nio.ByteBuffer;
-import java.nio.channels.FileChannel;
 import java.nio.channels.ReadableByteChannel;
+import java.nio.channels.WritableByteChannel;
+import java.util.List;
+import java.util.Optional;
+
+import org.neo4j.io.pagecache.PageCache;
+import org.neo4j.io.pagecache.PagedFile;
+import org.neo4j.kernel.impl.store.StoreType;
+
+import static java.nio.file.StandardOpenOption.CREATE;
+import static java.nio.file.StandardOpenOption.WRITE;
+import static org.neo4j.kernel.impl.store.StoreType.findStoreType;
+import static org.neo4j.kernel.impl.store.format.RecordFormat.NO_RECORD_SIZE;
 
 public class ToFileStoreWriter implements StoreWriter
 {
     private final File basePath;
     private final StoreCopyClient.Monitor monitor;
+    private final PageCache pageCache;
+    private final List<FileMoveAction> fileMoveActions;
 
-    public ToFileStoreWriter( File graphDbStoreDir, StoreCopyClient.Monitor monitor )
+    public ToFileStoreWriter( File graphDbStoreDir, StoreCopyClient.Monitor monitor, PageCache pageCache,
+            List<FileMoveAction> fileMoveActions )
     {
         this.basePath = graphDbStoreDir;
         this.monitor = monitor;
+        this.pageCache = pageCache;
+        this.fileMoveActions = fileMoveActions;
     }
 
     @Override
-    public long write( String path, ReadableByteChannel data, ByteBuffer temporaryBuffer,
-            boolean hasData ) throws IOException
+    public long write( String path, ReadableByteChannel data, ByteBuffer temporaryBuffer, boolean hasData,
+            int recordSize ) throws IOException
     {
         try
         {
             temporaryBuffer.clear();
             File file = new File( basePath, path );
-
             file.getParentFile().mkdirs();
+
+            String filename = file.getName();
+            final Optional<StoreType> storeType = findStoreType( filename );
+            final Optional<PagedFile> existingMapping = pageCache.getExistingMapping( file );
```

Comment:
I think it would be useful to have a comment explaining here what files we expect to be mapped, not mapped, not a stores to simplify future investigations